util/updateXML/sources/Dyd.py

```diff
@@ -0,0 +1,151 @@
+from .Model.ModelicaModel import ModelicaModel
+from .UpdateXMLExceptions import *
+
+
+class Dyd:
+    """
+    Represents a Dyd file
+
+    Attributes
+    ----------
+    filename_ : str
+        name of the Dyd file
+    job_parent_directory_ : str
+        absolute filepath of the job parent directory
+    dydtree_ : lxml.etree._ElementTree
+        lxml ElementTree of Dyd XML file
+    par_files_ : dict[str, Par]
+        par files content (key : par file path, value : par object)
+    bbms_ : list[BlackBoxModel]
+        list of blackBoxModels defined in Dyd XML file
+    modelica_models_ : list[ModelicaModel]
+        list of modelicaModel defined in Dyd XML file
+    curves_ : Curves
+        curves data of the job
+    final_state_values_ : FinalStateValues
+        final state values data of the job
+    """
+    def __init__(self, filepath, job_parent_directory, par_files, curves=None, final_state_values=None):
+        """
+        Parse the Dyd file
+
+        Parameters:
+            filepath (str): absolute filepath of the Dyd file to parse
+            job_parent_directory (str): absolute filepath of the job parent directory
+            par_files (dict[str, Par]): par files content (key : par file path, value : par object)
+            curves_ (Curves): curves data of the job
+        """
+        self.filename_ = os.path.basename(filepath)
+        self.job_parent_directory_ = job_parent_directory
+
+        xml_parser = lxml.etree.XMLParser(remove_comments=True)
+        try:
+            self.dydtree_ = lxml.etree.parse(filepath, xml_parser)
+        except OSError as exc:
+            raise DydFileDoesNotExist(filepath) from exc
+
+        self.par_files_ = par_files
+        self.curves_ = curves
+        self.final_state_values_ = final_state_values
+        self.bbms_ = list()
+        self.modelica_models_ = list()
+
+        root_element =  self.dydtree_.getroot()
+        for dyd_element in root_element:
+            if dyd_element.tag == xmlns(XML_BLACKBOXMODEL):
+                self.__parse_bbm_par_file(dyd_element)
+            elif dyd_element.tag == xmlns(XML_MODELICAMODEL):
+                self.__parse_modelica_model(dyd_element)
+            elif dyd_element.tag == xmlns(XML_CONNECT):
+                pass  # nothing to do
+            else:
+               raise UnknownDydElement(dyd_element.tag)
+
+    # ---------------------------------------------------------------
+    #   USER METHODS
+    # ---------------------------------------------------------------
+
+    def change_lib_name(self, current_lib_name, new_lib_name):
+        bbms = self.dydtree_.findall(xmlns(XML_BLACKBOXMODEL))
+        is_bbm_found = False
+        for bbm in bbms:
+            if bbm.attrib['lib'] == current_lib_name:
+                is_bbm_found = True
+                bbm.attrib['lib'] = new_lib_name
+        if not is_bbm_found:
+            raise LibraryNotFound(current_lib_name, self.dydtree_.docinfo.URL)
```

Comment:
If the library is not used in this case it is not an issue

Reply:
It may be convenient to raise an exception if the change_lib_name() method didn't find the library

Reply:
the plan is to launch all update to all pre existing jobs. If the script fails each time a library is not used it will be pretty hard to use